packages/core-users/src/module/configureUsersModule.ts

```diff
@@ -817,5 +852,14 @@ export const configureUsersModule = async ({
         {},
       );
     },
+    deleteUser: async ({ userId }, context) => {
+      const { modules } = context;
+      const { _id, ...user } = await findUserById(userId);
+      delete user?.services;
+      const maskedUserData = maskUserPropertyValues({ ...user, meta: null });
```

Comment:
This would lead to fields like updated and created having illegal values...

and generally speaking: we can't just set every property value to a sha hash because we can not know how that will effect the system that uses user extensions with custom code 😵‍💫

what if a hash of a username is beeing used as a username? this needs to be far more explicit.


this is an example of a user deletion extension from one customer
```
const moveUserToNirvana: VeloplusAuthAPI["moveUserToNirvana"] = async (
      userIdToDelete,
      loggedInUserId,
    ) => {
      await Users.updateOne(
        { _id: userIdToDelete },
        {
          $set: {
            username: `deleted-${new Date().getTime()}`,
            deleted: new Date(),
            deletedBy: loggedInUserId,
            emails: [],
            lastBillingAddress: {},
            services: {},
            roles: null,
            profile: null,
          },
        },
      );
    };
```


it should be similar to that: every default user property should be examined in detail and we should remove/unset/nullify all properties with PERSONAL DATA and only invent hashes for fields that are both required and have personal data (username beeing the only field). Properties that are purely metadata/informational and have no personal data value (for ex. an update date) can be left as is.